be/src/util/sm3.cpp

```diff
@@ -0,0 +1,241 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
```

Comment:
Is sm3 public domain ? or has any license?

Reply:
different sm3 implementation have there own's copyright. If we use it, we should keep the license as same as the source. And also add license to notice.txt.

Reply:
> Is sm3 public domain ? or has any license?

SM3 is a cryptographic hash function standard adopted by the government of the people's Republic of China.